torch/_dynamo/variables/tensor.py

```diff
@@ -807,6 +807,26 @@ def _register_hook_trampoline(tensor):
                 return tensor
 
             from .builder import wrap_fx_proxy
+            # TODO: figure out if this is necessary
+            if name =='to_local':
```

Comment:
wondering can we unify the logic with `from_local`? https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/variables/torch.py#L525

We can possibly see if we can make `to_local` be an TorchInGraphFunctionVariable too

Reply:
Turns out that change wasn't actually necessary, so I ended up removing it.

I add a different change though to handle hooks (see the PR update) - let me know if you think it's too boilerplate-y and worth refactoring with the to_local code!